tests/system-fetcher/tests/handler_test.go

```diff
@@ -238,6 +221,103 @@ func TestSystemFetcherSuccessForMoreThanOnePage(t *testing.T) {
+	}]`)
+
+	setMockSystems(t, mockSystems, tenant.TestTenants.GetDefaultTenantID())
+	setMockSystems(t, mockSystems, tenant.TestTenants.GetSystemFetcherTenantID())
+	defer cleanupMockSystems(t)
+
+	template, err := fixtures.CreateApplicationTemplateFromInput(t, ctx, certSecuredGraphQLClient, tenant.TestTenants.GetDefaultTenantID(), fixApplicationTemplate("temp1"))
+	defer fixtures.CleanupApplicationTemplate(t, ctx, certSecuredGraphQLClient, tenant.TestTenants.GetDefaultTenantID(), &template)
+	require.NoError(t, err)
+	require.NotEmpty(t, template.ID)
+
+	appTemplateInput2 := fixApplicationTemplate("temp2")
+	appTemplateInput2.Webhooks = append(appTemplateInput2.Webhooks, testPkg.BuildMockedWebhook(cfg.ExternalSvcMockURL+"/", directorSchema.WebhookTypeUnregisterApplication))
+	template2, err := fixtures.CreateApplicationTemplateFromInput(t, ctx, certSecuredGraphQLClient, tenant.TestTenants.GetDefaultTenantID(), appTemplateInput2)
+	defer fixtures.CleanupApplicationTemplate(t, ctx, certSecuredGraphQLClient, tenant.TestTenants.GetDefaultTenantID(), &template2)
+	require.NoError(t, err)
+	require.NotEmpty(t, template2.ID)
+
+	k8sClient, err := clients.NewK8SClientSet(ctx, time.Second, time.Minute, time.Minute)
+	require.NoError(t, err)
+	jobName := "system-fetcher-test"
+	namespace := "compass-system"
+	k8s.CreateJobByCronJob(t, ctx, k8sClient, "compass-system-fetcher", jobName, namespace)
+	defer func() {
+		k8s.PrintJobLogs(t, ctx, k8sClient, jobName, namespace, cfg.SystemFetcherContainerName, false)
+		k8s.DeleteJob(t, ctx, k8sClient, jobName, namespace)
+	}()
+
+	k8s.WaitForJobToSucceed(t, ctx, k8sClient, jobName, namespace)
+
+	description := "description"
+	baseUrl := "http://mainurl.com"
+	expectedApps := []directorSchema.ApplicationExt{
+		{
+			Application: directorSchema.Application{
+				Name:                  "name1",
+				Description:           &description,
+				BaseURL:               &baseUrl,
+				ApplicationTemplateID: &template.ID,
+				SystemNumber:          str.Ptr("1"),
+			},
+			Labels: applicationLabels("name1", true),
+		},
+		{
+			Application: directorSchema.Application{
+				Name:         "name2",
+				Description:  &description,
+				BaseURL:      &baseUrl,
+				SystemNumber: str.Ptr("2"),
+			},
+			Labels: applicationLabels("name2", false),
+		},
+	}
+
+	resp, actualApps := retrieveAppsForTenant(t, ctx, tenant.TestTenants.GetDefaultTenantID())
+	defer func() {
+		for _, app := range resp.Data {
+			fixtures.CleanupApplication(t, ctx, certSecuredGraphQLClient, tenant.TestTenants.GetDefaultTenantID(), app)
+		}
+	}()
+	require.ElementsMatch(t, expectedApps, actualApps)
+
+	resp, actualApps = retrieveAppsForTenant(t, ctx, tenant.TestTenants.GetSystemFetcherTenantID())
+	defer func() {
+		for _, app := range resp.Data {
+			fixtures.CleanupApplication(t, ctx, certSecuredGraphQLClient, tenant.TestTenants.GetSystemFetcherTenantID(), app)
+		}
+	}()
+	require.ElementsMatch(t, expectedApps, actualApps)
+}
```

Comment:
To be 100% sure that the two systems are one and the same I would additionally check that the IDs of the apps from the first request are the same as the IDs of the apps from the second request.